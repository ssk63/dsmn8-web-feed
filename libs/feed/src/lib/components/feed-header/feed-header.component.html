<div class="feed-header">
  <div class="feed-header__details">
    <h1>Help promote DSMN8's post on Linkedin</h1>
  </div>
  <p>
    {{ text }}
  </p>
</div>
