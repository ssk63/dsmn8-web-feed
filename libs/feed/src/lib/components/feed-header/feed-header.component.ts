import { Component, Input, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'dsmn8-feed-header',
  templateUrl: './feed-header.component.html',
  styleUrls: ['./feed-header.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class FeedHeaderComponent {
  @Input() text: string | undefined;
}
