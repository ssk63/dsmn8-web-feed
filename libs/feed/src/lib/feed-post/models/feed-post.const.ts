export const likeBtnLabel = 'Like on LinkedIn (5 pts)';
export const reshareBtnLabel = 'Reshare on LinkedIn (5 pts)';
