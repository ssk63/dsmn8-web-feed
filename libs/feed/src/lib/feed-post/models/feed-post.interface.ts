/**
 * Interface for the 'FeedPost' data
 */
export interface FeedPostEntity {
  id: string | number;
  name: string;
  content: string;
  text: string;
  boostPostMetrics: IMetrics;
  imgUrl: string;
}

export interface IMetrics {
  likes: number;
  shares: number;
  totalReach: number;
}
