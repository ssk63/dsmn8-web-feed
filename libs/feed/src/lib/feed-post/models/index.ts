export * from './feed-post.interface';
export * from './feed-post.mock-data';
