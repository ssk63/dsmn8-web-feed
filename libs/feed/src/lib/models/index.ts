export * from './feed.interface';
export * from './feed.mock.data';
export * from './feed.const';
