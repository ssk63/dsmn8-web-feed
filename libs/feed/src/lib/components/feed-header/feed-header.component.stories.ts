import { moduleMetadata, Story, Meta } from '@storybook/angular';
import { mockFeed } from '../../models';
import { FeedHeaderComponent } from './feed-header.component';

export default {
  title: 'Pages/Feed/FeedHeaderComponent',
  component: FeedHeaderComponent,
  decorators: [
    moduleMetadata({
      imports: [],
    }),
  ],
  argTypes: {
    text: {
      name: 'text',
      description: 'Sets the header text.',
      type: { name: 'string' },
      control: {
        type: 'text',
      },
    },
  },
  parameters: {
    jest: 'feed-header.component',
  }
} as Meta<FeedHeaderComponent>;

const Template: Story<FeedHeaderComponent> = (args: FeedHeaderComponent) => ({
  props: args,
});

export const HeaderText = Template.bind({});
HeaderText.args = {
  text: mockFeed[0].text,
};
