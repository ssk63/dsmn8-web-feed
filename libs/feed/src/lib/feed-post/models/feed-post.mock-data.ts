import { FeedPostEntity } from './feed-post.interface';

export const mockFeedPosts: FeedPostEntity[] = [
  {
    content: `FAILED previously with employee advocacy?

    It happens! 🤷‍♀️
    
    There are a number of reasons why this might happen.
    
    It could be that the program was poorly run by your predecessor.
    
    They could have chosen a tech partner that wasn’t right for your company.
    
    Maybe employees weren’t clued up on what was in it for them!
    
    Well, if you’re looking to give it another go…
    
    Because you KNOW what’s possible with an employee advocacy program and you’re determined to get those results?!
    
    LOVE. IT. 👏
    
    And you’re in luck!
    
    Here at DSMN8, we’ve been helping launch and run hundreds of employee advocacy programs since 2016, so we have a few tricks up our sleeves!
    
    We’ve learned a thing or two about the reasons why an employee advocacy program can go stale, and we’ve put together a guide to getting things up and running a second time.
    
    Check out the link in the comments to learn how to reboot and win with employee advocacy.
    
    Let’s get set up for success 🔗👇`,
    name: 'DSMN8 - The Employee Influence Platform',
    id: '1664896526',
    text: 'By liking and re-sharing this post, you can help boost the organic reach of the content',
    boostPostMetrics: {
      likes: 2,
      shares: 0,
      totalReach: 5307,
    },
  },
  {
    content: `LET ME SET THE SCENE

    You’re running an employee advocacy program…
    
    It’s going well so far.
    
    And you’re looking to scale it up. 🙌
    
    Or perhaps it’s not working as well as you thought it would…
    
    And you’re trying to figure out why. 😫
    
    Well, you’re in luck.
    
    We recently covered how to improve employee advocacy.
    
    Everything from how to encourage employee engagement to how to fix common employee advocacy mistakes.
    
    Let’s go! 🔗👇
    
    (link in the comments!)`,
    name: 'DSMN8 - The Employee Influence Platform',
    id: '1664885288',
    text: 'By liking and re-sharing this post, you can help boost the organic reach of the content',
    boostPostMetrics: {
      likes: 6,
      shares: 0,
      totalReach: 10345,
    },
  },
];
