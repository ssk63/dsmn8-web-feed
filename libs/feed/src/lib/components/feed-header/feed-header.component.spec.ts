import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { mockFeed } from '../../models';
import { FeedHeaderComponent } from './feed-header.component';

describe('FeedHeaderComponent', () => {
  let component: FeedHeaderComponent;
  let fixture: ComponentFixture<FeedHeaderComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FeedHeaderComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(FeedHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should render header text when the text is exists', () => {
    // arrange
    component.text = mockFeed[0].text;
    // act
    fixture.detectChanges();
    const feedHeader = fixture.debugElement.query(
      By.css('.feed-header')
    ).nativeElement;
    // assert
    expect(feedHeader).toMatchSnapshot();
  });
});
