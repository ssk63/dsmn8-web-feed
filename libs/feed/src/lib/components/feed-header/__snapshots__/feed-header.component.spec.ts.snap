// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedHeaderComponent should render header text when the text is exists 1`] = `
<div
  class="feed-header"
>
  <div
    class="feed-header__details"
  >
    <h1>
      Help promote DSMN8's post on Linkedin
    </h1>
  </div>
  <p>
     By liking and re-sharing this post, you can help boost the organic reach of the content 
  </p>
</div>
`;
